src/main/java/org/reaktivity/rym/internal/RymInstall.java

```diff
@@ -0,0 +1,301 @@
+        contents.append("    </cache>\n");
+        contents.append("  </caches>\n");
+        contents.append("  <resolvers>\n");
+        contents.append("    <chain name=\"default\">\n");
+        contents.append("      <url name=\"blah\">\n");
+        contents.append("        <ivy pattern=\"https://repo1.maven.org/maven2/[module]/[revision]/ivy-[revision].xml\" />\n");
+        contents.append("        <artifact pattern=\"https://repo1.maven.org/maven2/[module]/[revision]/[artifact]-[revision].[ext]\" />");
+        contents.append("      </url>\n");
+//        contents.append("      <ibiblio name=\"central\" m2compatible=\"true\" cache=\"mycache\" />\n");
+        contents.append("    </chain>\n");
+        contents.append("  </resolvers>\n");
+        contents.append("</ivysettings>\n");
+
+        BufferedWriter out = new BufferedWriter(new FileWriter(ivySettingsFile));
+        out.write(contents.toString());
+        out.close();
+        return ivySettingsFile;
+    }
+
+    private File createIvyFile() throws IOException
+    {
+        File ivyFile = File.createTempFile("ivy.xml", null);
+        ivyFile.deleteOnExit();
+
+        // TODO What to put for "organization" and "module" values here:
+        StringBuffer contents = new StringBuffer();
+        contents.append("<ivy-module version=\"2.0\">\n");
+        contents.append("  <info organisation=\"reaktivity\" module=\"reaktivity-deps\"/>\n");
+        contents.append("  <dependencies>\n");
+        for (Map.Entry<String, String> entry : dependencies.entrySet())
+        {
+            String[] dependencyComponents = entry.getKey().split(":");
+            contents.append(String.format("    <dependency org=\"%s\" name=\"%s\" rev=\"%s\"/>\n",
+                dependencyComponents[0],
+                dependencyComponents[1],
+                entry.getValue()));
+        }
+        contents.append("    <dependency org=\"commons-lang\" name=\"commons-lang\" rev=\"2.0\"/>\n");
+        contents.append("    <dependency org=\"commons-cli\" name=\"commons-cli\" rev=\"1.2\"/>\n");
+        contents.append("  </dependencies>\n");
+        contents.append("</ivy-module>\n");
+
+        BufferedWriter out = new BufferedWriter(new FileWriter(ivyFile));
+        out.write(contents.toString());
+        out.close();
+        return ivyFile;
+    }
+
+    private void resolveDependencies() throws ParseException, IOException
+    {
+        IvySettings ivySettings = new IvySettings();
+        org.apache.ivy.Ivy ivy = Ivy.newInstance(ivySettings);
+        ivy.configure(createIvySettingsFile());
+
+        ResolveOptions resolveOptions = new ResolveOptions();
+        resolveOptions.setLog(LOG_QUIET); // TODO Pick logging option: Always, on download only, or never.
+        resolveOptions.setOutputReport(false); // TODO Show the report? LOG_QUIET suppresses it, even if setOutputReport is true
+        ResolveReport resolveReport = ivy.resolve(createIvyFile(), resolveOptions);
```

Comment:
To eliminate the need for `ivy.xml` file, we'll need to use one of the overloaded `ivy.resolve(...)` methods with different parameters.

For example, [`Ivy.resolve(ModuleDescriptor md, ResolveOptions options)`](https://github.com/apache/ant-ivy/blob/master/src/java/org/apache/ivy/Ivy.java#L518).

A [`ModuleDescriptor`](https://github.com/apache/ant-ivy/blob/master/src/java/org/apache/ivy/core/module/descriptor/ModuleDescriptor.java) can be created via [`DefaultModuleDescriptor.newDefaultInstance(ModuleRevisionId)`](https://github.com/apache/ant-ivy/blob/master/src/java/org/apache/ivy/core/module/descriptor/DefaultModuleDescriptor.java#L65).

A `ModuleRevisionId` can be created by [`ModuleRevisionId.newInstance(String organisation, String name, String revision)`](https://github.com/apache/ant-ivy/blob/master/src/java/org/apache/ivy/core/module/id/ModuleRevisionId.java#L95) for example.

Each `ModuleRevisionId` seems like it would be a one-to-one match for each of the listed dependencies.